src/main/java/seedu/address/model/property/Property.java

```diff
@@ -0,0 +1,124 @@
+package seedu.address.model.property;
+
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Property in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Property {
+
+    // Identity fields
+    private final Name name;
+    private final Price price;
+
+    // Data fields
+    private final Address address;
+    private final Description description;
+    private final Set<Tag> tags = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Property(Name name, Price price, Address address, Description description, Set<Tag> tags) {
+        requireAllNonNull(name, price, address, description, tags);
+        this.name = name;
+        this.price = price;
+        this.address = address;
+        this.description = description;
+        this.tags.addAll(tags);
+    }
+
+    public Name getName() {
+        return name;
+    }
+
+    public Address getAddress() {
+        return address;
+    }
+
+    public Price getPrice() {
+        return price;
+    }
+    public Description getDescription() {
+        return description;
+    }
+
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    /**
+     * Returns true if both properties have the same name and price.
+     * This defines a weaker notion of equality between two properties.
+     */
+    public boolean isSameProperty(Property otherProperty) {
+        if (otherProperty == this) {
+            return true;
+        }
+
+        return otherProperty != null
+                && otherProperty.getName().equals(getName())
+                && otherProperty.getPrice().equals(getPrice());
+    }
+
+    /**
+     * Returns true if both persons have the same identity and data fields.
+     * This defines a stronger notion of equality between two persons.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        if (!(other instanceof Property)) {
+            return false;
+        }
+
+        Property otherProperty = (Property) other;
+        return otherProperty.getName().equals(getName())
+                && otherProperty.getPrice().equals(getPrice())
+                && otherProperty.getAddress().equals(getAddress())
+                && otherProperty.getDescription().equals(getDescription())
+                && otherProperty.getTags().equals(getTags());
```

Comment:
We may not need to compare tags for properties

Reply:
Ok, I deleted the tags part! Thanks Ian 👍